hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/kafka/PropertiesResolver.java

```diff
@@ -0,0 +1,110 @@
+
+import com.hazelcast.jet.sql.impl.connector.SqlConnector;
+import org.apache.kafka.common.serialization.ByteArrayDeserializer;
+import org.apache.kafka.common.serialization.ByteArraySerializer;
+import org.apache.kafka.common.serialization.DoubleDeserializer;
+import org.apache.kafka.common.serialization.DoubleSerializer;
+import org.apache.kafka.common.serialization.FloatDeserializer;
+import org.apache.kafka.common.serialization.FloatSerializer;
+import org.apache.kafka.common.serialization.IntegerDeserializer;
+import org.apache.kafka.common.serialization.IntegerSerializer;
+import org.apache.kafka.common.serialization.LongDeserializer;
+import org.apache.kafka.common.serialization.LongSerializer;
+import org.apache.kafka.common.serialization.ShortDeserializer;
+import org.apache.kafka.common.serialization.ShortSerializer;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.apache.kafka.common.serialization.StringSerializer;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+import static com.hazelcast.jet.sql.impl.connector.SqlConnector.AVRO_FORMAT;
+import static com.hazelcast.jet.sql.impl.connector.SqlConnector.JAVA_FORMAT;
+import static com.hazelcast.jet.sql.impl.connector.SqlConnector.JSON_FORMAT;
+import static com.hazelcast.jet.sql.impl.connector.SqlConnector.OPTION_KEY_FORMAT;
+import static com.hazelcast.jet.sql.impl.connector.SqlConnector.OPTION_VALUE_FORMAT;
+
+final class PropertiesResolver {
+
+    static final String KEY_SERIALIZER = "key.serializer";
+    static final String KEY_DESERIALIZER = "key.deserializer";
+    static final String VALUE_SERIALIZER = "value.serializer";
+    static final String VALUE_DESERIALIZER = "value.deserializer";
+
+    static final String AVRO_SERIALIZER = "io.confluent.kafka.serializers.KafkaAvroSerializer";
+    static final String AVRO_DESERIALIZER = "io.confluent.kafka.serializers.KafkaAvroDeserializer";
+
+    static final String JSON_SERIALIZER = ByteArraySerializer.class.getCanonicalName();
+    static final String JSON_DESERIALIZER = ByteArrayDeserializer.class.getCanonicalName();
+
+    private PropertiesResolver() {
+    }
+
+    static Properties resolveProperties(Map<String, String> options) {
+        Properties properties = new Properties();
+        properties.putAll(options);
+
+        properties(options, true).forEach(properties::putIfAbsent);
+        properties(options, false).forEach(properties::putIfAbsent);
+
+        return properties;
+    }
+
+    private static Map<String, String> properties(Map<String, String> options, boolean isKey) {
+        Map<String, String> properties = new HashMap<>();
+
+        String serializer = isKey ? KEY_SERIALIZER : VALUE_SERIALIZER;
+        String deserializer = isKey ? KEY_DESERIALIZER : VALUE_DESERIALIZER;
+
+        String format = options.get(isKey ? OPTION_KEY_FORMAT : OPTION_VALUE_FORMAT);
+        if (JAVA_FORMAT.equals(format)) {
```

Comment:
I think we can use `switch (format)`. Since java 7 it works for strings.

Reply:
I would like to tolerate `null`s